src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/LazyExtension.cs

```diff
@@ -78,17 +68,12 @@ public LazyExtension(TestPluginInformation pluginInfo, Type metadataType)
         /// <param name="metadata">Test extension metadata</param>
         public LazyExtension(TestPluginInformation pluginInfo, TMetadata metadata)
         {
-            if (pluginInfo == null)
-            {
-                throw new ArgumentNullException("pluginInfo");
-            }
-
             if (metadata == null)
             {
                 throw new ArgumentNullException("metadata");
             }
 
-            this.testPluginInfo = pluginInfo;
+            this.testPluginInfo = pluginInfo ?? throw new ArgumentNullException("pluginInfo");
```

Comment:
Can you switch to `nameof` wherever applicable?

```suggestion
            this.testPluginInfo = pluginInfo ?? throw new ArgumentNullException(nameof(pluginInfo));
```

Reply:
Added where found